app/src/ext_power_generic.c

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2020 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#define DT_DRV_COMPAT zmk_ext_power_generic
+
+#include <device.h>
+#include <init.h>
+#include <drivers/gpio.h>
+#include <drivers/ext_power.h>
+
+#if DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT)
+
+#include <logging/log.h>
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+struct ext_power_generic_config {
+    const char *label;
+    const u8_t pin;
+    const u8_t flags;
+};
+
+struct ext_power_generic_data {
+    struct device *gpio;
+    bool status;
+};
+
+static int ext_power_generic_enable(struct device *dev)
+{
+    struct ext_power_generic_data *data = dev->driver_data;
+    const struct ext_power_generic_config *config = dev->config_info;
+
+    if (gpio_pin_set(data->gpio, config->pin, GPIO_OUTPUT_HIGH)) {
+        LOG_DBG("Failed to set ext-power control pin");
+        return -EIO;
+    }
+    data->status = true;
+    return 0;
+}
+
+static int ext_power_generic_disable(struct device *dev)
+{
+    struct ext_power_generic_data *data = dev->driver_data;
+    const struct ext_power_generic_config *config = dev->config_info;
+
+    if (gpio_pin_set(data->gpio, config->pin, GPIO_OUTPUT_LOW)) {
+        LOG_DBG("Failed to clear ext-power control pin");
+        return -EIO;
+    }
+    data->status = false;
+    return 0;
+}
+
+static int ext_power_generic_get(struct device *dev, bool *status)
+{
+    struct ext_power_generic_data *data = dev->driver_data;
+
+    *status = data->status;
+    return 0;
+}
+
+static int ext_power_generic_init(struct device *dev)
+{
+    struct ext_power_generic_data *data = dev->driver_data;
+    const struct ext_power_generic_config *config = dev->config_info;
+
+    data->gpio = device_get_binding(config->label);
+    if (data->gpio == NULL) {
+        LOG_ERR("Failed to get ext-power control device");
+        return -EINVAL;
+    }
+
+    if (gpio_pin_configure(data->gpio, config->pin, config->flags | GPIO_OUTPUT)) {
+        LOG_DBG("Failed to configure ext-power control pin");
+        return -EIO;
+    }
+
+    // Enable the VCC Output on init
+    if (gpio_pin_set(data->gpio, config->pin, GPIO_OUTPUT_HIGH)) {
```

Comment:
I have mixed feelings about just doing this on init. I'd rather move this to the *application* init. Thoughts?